sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -2795,3 +2795,69 @@ case class Sentences(
     copy(str = newFirst, language = newSecond, country = newThird)
 
 }
+
+/**
+ * Splits a given string by a specified delimiter.
+ */
+case class SplitByDelimiter(
+    str: Expression,
+    delimiter: Expression)
+  extends BinaryExpression with ImplicitCastInputTypes with NullIntolerant {
+  override def dataType: DataType = ArrayType(StringType, containsNull = false)
+  override def inputTypes: Seq[DataType] = Seq(StringType, StringType)
+  override def left: Expression = str
+  override def right: Expression = delimiter
+
+  override def nullSafeEval(string: Any, delimiter: Any): Any = {
+    val strings = string.asInstanceOf[UTF8String].split(
+      delimiter.asInstanceOf[UTF8String], -1, true)
+    new GenericArrayData(strings.asInstanceOf[Array[Any]])
+  }
+
+  override def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
+    val arrayClass = classOf[GenericArrayData].getName
+    nullSafeCodeGen(ctx, ev, (str, delimiter) => {
+      // Array in java is covariant, so we don't need to cast UTF8String[] to Object[].
+      s"""${ev.value} = new $arrayClass($str.split($delimiter,-1, true));""".stripMargin
+    })
+  }
+
+  override protected def withNewChildrenInternal(
+    newFirst: Expression, newSecond: Expression): SplitByDelimiter =
+    copy(str = newFirst, delimiter = newSecond)
+}
+
+/**
+ * Splits a given string by a specified delimiter and returns the requested part.
+ * If any input is null, or index is out of range of split parts, returns null.
+ */
+@ExpressionDescription(
+  usage = """
+    _FUNC_(str, delimiter, partNum) - Splits `str` by delimiter and return
+      requested part of the split (1-based).
```

Comment:
Can we mention if negative partNum is supported or not?

Reply:
added explanation on negative partNum and what is usage of that.